src/index.ts

```diff
@@ -40,10 +40,26 @@ export const go = async <T, E extends Error>(fn: Promise<T> | (() => Promise<T>)
     if (typeof fn === 'function') {
       return fn()
         .then(success)
-        .catch((err) => createError(err));
+        .catch((err) => createGoError(err));
     }
-    return fn.then(success).catch((err) => createError(err));
+    return fn.then(success).catch((err) => createGoError(err));
   } catch (err) {
-    return createError(err);
+    return createGoError(err);
   }
 };
+
+// NOTE: This needs to be written using 'function' syntax (cannot be arrow function)
+// See: https://github.com/microsoft/TypeScript/issues/34523#issuecomment-542978853
+export function assertGoSuccess<T>(result: GoResult<T>): asserts result is GoResultSuccess<T> {
+  if (result.success) {
+    throw result.data;
```

Comment:
I'm confused by this - why does it throw `result.data` if `result.success` is truthy? Shouldn't it throw if `result.success` is falsey?

It's also exactly the same as the `assertGoError` function (apart from the function types)?

Reply:
Because there is a bug :) I am missing the `success` property as @amarthadan correctly pointed out.

This was painful, because I never asserted the exact response from go utils (I always used the `fail` and `success` methods because of the jest.toEqual limitation )